api/src/main/java/gov/va/vro/api/model/ClaimProcessingException.java

```diff
@@ -1,14 +1,34 @@
 package gov.va.vro.api.model;
 
 import lombok.Getter;
-import lombok.RequiredArgsConstructor;
 import org.springframework.http.HttpStatus;
 
-@RequiredArgsConstructor
-@Getter
 public class ClaimProcessingException extends Exception {
 
-  private final String claimSubmissionId;
-  private final HttpStatus httpStatus;
-  private final String message;
+  @Getter private final String claimSubmissionId;
+  @Getter private final HttpStatus httpStatus;
+
+  public ClaimProcessingException(String claimSubmissionId, HttpStatus httpStatus, String message) {
+    super(message);
+    this.claimSubmissionId = claimSubmissionId;
+    this.httpStatus = httpStatus;
+  }
+
+  public ClaimProcessingException(
+      String claimSubmissionId, HttpStatus httpStatus, Exception exception) {
+    super(exception);
+    this.claimSubmissionId = claimSubmissionId;
+    this.httpStatus = httpStatus;
+  }
+
+  public String getOriginalMessage() {
```

Comment:
It is not really "original" message, is it? Could the code be a bit simpler? But it is okay if there is a use for it.

Reply:
The message that was displayed before was a generic exception saying that something occured in exchange XYZ. Going one level down in the message sequence reveals the actual reason for the exception, which in this case was a timeout. Perhaps in the future I will store the entire stacktrace as string if necessary.